### PR TITLE
Refactor histogram to gauge and more test examples

### PR DESCRIPTION
**Pull request by abtris**

### CODE REVIEW

**Comment by abtris** [comment]
@w-vi still tests failing, looks as problem with order, but I'm not sure. Pls look if find something more

File: server_test.go [hunk 1]
```diff
@@ -42,13 +42,18 @@ var fullTests = []struct {
 		},
 	},
 	{
-		cnt: 3,
-		Req: `222 <134>1 2015-04-07T16:01:43.517062+00:00 host app web.1 - info: responseLogger: metric#tag#route=/parser metric#request_id=11747467-f4ce-4b06-8c99-92be968a02e3 metric#request_length=541 metric#response_length=5163 metric#parser_time=5ms metric#eventloop.avg_ms=1.589123`,
+		cnt: 8,
```
Comment (w-vi) [inline server_test.go]: There are 9 test cases if I count correctly, that's why it fails

Comment (w-vi) [reply server_test.go]: From the Travis CI log : 
`server_test.go:180: Count of expected results isn't equal to inputs`

**Comment by abtris** [comment]
@w-vi fixed

**Pull request merged**